src/ucp/proto/proto_common.c

```diff
@@ -676,6 +677,16 @@ ucp_proto_common_find_lanes(const ucp_proto_init_params_t *params,
             }
         }
 
+        /* The two devices must also have internal reachability */
+        lane_sys_dev = context->tl_rscs[rsc_index].tl_rsc.sys_device;
+        if (!ucs_topo_is_memory_reachable(lane_sys_dev,
+                                          select_param->sys_dev)) {
+            ucs_trace("%s: no reachability between lane_sys_dev=%u and "
+                      "sys_dev=%u",
+                      lane_desc, lane_sys_dev, select_param->sys_dev);
```

Comment:
I think it is better to put:
```C
ucs_trace(sys_dev[%u] cannot reach memory of sys_dev[%u]", device, mem_device);
```
to `ucs_topo_is_memory_reachable`, and remove such trace from this function, and from `ucp_memh_sys_dev_reachable`.